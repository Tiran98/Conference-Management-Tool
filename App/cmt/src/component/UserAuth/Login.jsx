import React, { useState, useEffect, useRef, useLayoutEffect } from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import { Radio, RadioGroup, FormLabel, TextField, FormControlLabel, Paper, Avatar, Button, CssBaseline, Grid, Typography, Container, Divider } from '@material-ui/core/';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import { useHistory } from 'react-router-dom';
import { useForm, Controller } from "react-hook-form";
import { withStyles } from '@material-ui/core/styles';

import useStyles from './styles';

const Login = ({ setDrawerState }) => {
    const classes = useStyles();
    const { control, handleSubmit, reset } = useForm();
    const [userType, setUserType] = useState("attendee");
    const [userToken, setUserToken] = useState();
    const [userProfile, setUserProfile] = useState([]);
    const [formData, setFormData] = useState([]);
    const isFirstRender = useRef(true);
    const history = useHistory();

    const CssTextField = withStyles({
        root: {
          '& .MuiInputLabel-root': {
            color: '#a3a3a3',
          },
          '& .MuiTextField-root': {
            color: '#a3a3a3',
          },
          '& .MuiFormHelperText-root': {
            color: '#a3a3a3',
          },
          '& label.Mui-focused': {
            color: '#a3a3a3',
          },
          '& .MuiInputBase-input':{
            color: '#ffffff',
          },
          '& .MuiInput-underline:after': {
            borderBottomColor: '#a3a3a3',
          },
          '& .MuiOutlinedInput-root': {
            '& fieldset': {
              borderColor: '#a3a3a3',
            },
            '&:hover fieldset': {
              borderColor: '#cccccc',
            },
            '&.Mui-focused fieldset': {
              borderColor: '#FFB101',
            },
          },
        },
        // input: {
        //   color: "white"
        // }
    })(TextField);

    useEffect(() => {
        handleDrawerClose();

        if (isFirstRender.current) {
          isFirstRender.current = false // toggle flag after first render/mounting
          return;
        }

        if(userType == 'attendee' || userType == 'researcher' || userType == 'workshop_presenter') {
          submitForm(formData);
        }

        localStorage.setItem('userToken', userToken);

    }, [formData, userToken]);


    useEffect(() => {
      localStorage.setItem('profile', JSON.stringify(userProfile));
      localStorage.setItem('userType', JSON.stringify(userType));
    }, [userProfile])

    const onSubmit = (data) => {
    
      setFormData({
          email : data.email,
          password : data.password,
          userType : userType
      })

      // submitForm(formData);
    }

    const submitForm = (data) => {
      // console.log(data);

      axios.post('http://localhost:5000/api/user/login',
      {
        email : data.email,
        password : data.password,
        userType : data.userType

      }). then((response) => {
        setUserToken(response.data.token);
        setUserProfile(response.data.user);
        history.push('/');
      }).catch((err) => {
        console.log(err);
      })

    }

    // const fetchUser = async() => {
    //   const response = await fetch('http://localhost:5001/vehicles/');
    //   const data = await response.json();
    //   setVehicles(data);
    // }

    const handleDrawerClose = () => {
        setDrawerState(false);
    };

    const handleRadioChange = (event) => {
        setUserType(event.target.value, console.log(userType));
    };

    
    return (
        <div className={classes.body}>
        <Container component="main" maxWidth="sm">
            <CssBaseline />
            <div className={classes.paper}>
                <Avatar className={classes.avatar}>
                    <LockOutlinedIcon />
                </Avatar>
                <Typography component="h1" variant="h5" gutterBottom>Sign In</Typography>
                <form onSubmit={handleSubmit(onSubmit)} className={classes.form}>
                    <Grid container spacing={2}>
                        <Grid item xs={12} sm={12}>
                            <Controller
                                name="email"
                                control={control}
                                defaultValue=""
                                render={({ field }) => 
                                <CssTextField fullWidth label="Email" variant="outlined" color="primary" {...field} />}
                            />
                        </Grid> 
                        <Grid item xs={12} sm={12}>
                            <Controller
                                name="password"
                                control={control}
                                defaultValue=""
                                render={({ field }) => 
                                <CssTextField fullWidth label="Password" variant="outlined" color="primary" {...field} />}
                            />
                        </Grid>
                        <Grid item xs={12} sm={12}>
                            <FormLabel component="legend" className={classes.radioGoupLabel}>Account Type</FormLabel>
                            <RadioGroup aria-label="user_type" name="user_type" row value={userType} onChange={handleRadioChange}>
                                <FormControlLabel value="attendee" control={<Radio className={classes.radioGoup} />} label="Attendee" className={classes.radioGoup} />
                                <FormControlLabel value="researcher" control={<Radio className={classes.radioGoup} />} label="Researcher" className={classes.radioGoup} />
                                <FormControlLabel value="workshop_presenter" control={<Radio className={classes.radioGoup} />} label="Workshop Presenter" className={classes.radioGoup} />
                            </RadioGroup>
                        </Grid>     
                    </Grid>
                    <Button type="submit" fullWidth variant="contained" color="secondary" className={classes.submit}>
                       Sign In
                    </Button>
                    <Grid container justify="flex-end">
                        <Grid item>
                            <Button component={Link} to="/register" size="small" style={{ color: '#ffffff' }}>
                                Don't have an account? Sign Up.
                            </Button>
                        </Grid>
                    </Grid>
                </form>
            </div>
            </Container>
           
        </div>
    )
}

export default Login;
