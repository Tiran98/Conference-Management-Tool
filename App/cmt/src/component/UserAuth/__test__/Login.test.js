import React from 'react';
import ReactDOM from 'react-dom';
import Login from '../Login';
import { BrowserRouter as Router } from 'react-router-dom';
import { isTSAnyKeywod } from '@babel/types';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import TestRenderer from 'react-test-renderer';

afterEach(cleanup);
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Router><Login></Login></Router>, div);
});

// it("renders login correctly", () => {
//     render(<Router><Login></Login></Router>);
// });

it("matches snapshot", () => {
    const tree = TestRenderer.create(<Router><Login></Login></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});