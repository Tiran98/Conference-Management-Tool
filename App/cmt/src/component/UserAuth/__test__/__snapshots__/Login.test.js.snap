// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <main
    className="MuiContainer-root MuiContainer-maxWidthSm"
  >
    <div
      className="makeStyles-paper-2"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-7 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Sign In
      </h1>
      <form
        className="makeStyles-form-8"
        onSubmit={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-28 MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
              >
                Email
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="email"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-19"
                  >
                    <span>
                      Email
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-28 MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
              >
                Password
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="password"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-19"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
          >
            <legend
              className="MuiFormLabel-root makeStyles-radioGoupLabel-14"
            >
              Account Type
            </legend>
            <div
              aria-label="user_type"
              className="MuiFormGroup-root MuiFormGroup-row"
              role="radiogroup"
            >
              <label
                className="MuiFormControlLabel-root makeStyles-radioGoup-13"
              >
                <span
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-21 MuiRadio-root MuiRadio-colorSecondary makeStyles-radioGoup-13 PrivateSwitchBase-checked-22 Mui-checked MuiIconButton-colorSecondary"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <input
                      checked={true}
                      className="PrivateSwitchBase-input-24"
                      name="user_type"
                      onChange={[Function]}
                      type="radio"
                      value="attendee"
                    />
                    <div
                      className="PrivateRadioButtonIcon-root-25 PrivateRadioButtonIcon-checked-27"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-26"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </div>
                  </span>
                </span>
                <span
                  className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  Attendee
                </span>
              </label>
              <label
                className="MuiFormControlLabel-root makeStyles-radioGoup-13"
              >
                <span
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-21 MuiRadio-root MuiRadio-colorSecondary makeStyles-radioGoup-13 MuiIconButton-colorSecondary"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <input
                      checked={false}
                      className="PrivateSwitchBase-input-24"
                      name="user_type"
                      onChange={[Function]}
                      type="radio"
                      value="researcher"
                    />
                    <div
                      className="PrivateRadioButtonIcon-root-25"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-26"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </div>
                  </span>
                </span>
                <span
                  className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  Researcher
                </span>
              </label>
              <label
                className="MuiFormControlLabel-root makeStyles-radioGoup-13"
              >
                <span
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-21 MuiRadio-root MuiRadio-colorSecondary makeStyles-radioGoup-13 MuiIconButton-colorSecondary"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <input
                      checked={false}
                      className="PrivateSwitchBase-input-24"
                      name="user_type"
                      onChange={[Function]}
                      type="radio"
                      value="workshop_presenter"
                    />
                    <div
                      className="PrivateRadioButtonIcon-root-25"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-26"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </div>
                  </span>
                </span>
                <span
                  className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  Workshop Presenter
                </span>
              </label>
            </div>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-12 MuiButton-containedSecondary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Sign In
          </span>
        </button>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <a
              aria-disabled={false}
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
              href="/register"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              style={
                Object {
                  "color": "#ffffff",
                }
              }
              tabIndex={0}
            >
              <span
                className="MuiButton-label"
              >
                Don't have an account? Sign Up.
              </span>
            </a>
          </div>
        </div>
      </form>
    </div>
  </main>
</div>
`;
